File: services/faturamento-api/Entities/Nota.cs
namespace faturamento_api.Entities;

public class Nota
{
    public int Id { get; set; }
    public int Number { get; set; }
    public string Status { get; set; } = "Aberta";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public List<ItemNota> Items { get; set; } = [];
}

File: frontend/src/app/models/nota.ts
export interface ItemNota {
  id: number;
  invoiceId: number;
  productId: number;
  productCode: string;
  productDescription: string;
  quantity: number;
}

export interface Nota {
  id: number;
  number: number;
  status: string;
  createdAt: string;
  items: ItemNota[];
}